src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,118 @@
+import clsx from 'clsx';
+import React from 'react';
+import Icon from '../Icon';
+import styles from './Avatar.module.css';
+
+export type UserData = {
+  /**
+   * The full name of the attached user (e.g., Jane Doe, David S. Pumpkins)
+   */
+  fullName: string;
+  /**
+   * User ID associated with the attached user
+   */
+  id?: string | number;
+  /**
+   * Additional data for an attached user (email, etc.)
+   */
+  [k: string]: string | number | undefined;
+};
+
+export type AvatarElement = 'div' | 'button';
+
+export interface Props {
+  /**
+   * Label for the given avatar. Defaults to a string using user data.
+   */
+  ariaLabel: string;
+  as?: AvatarElement;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * The shape of the avatar
+   */
+  shape?: 'circle' | 'square';
+  /**
+   * The size of the component
+   */
+  size?: 'sm' | 'md' | 'lg';
+  /**
+   * The URL to an image resource (loaded lazily)
+   */
+  src?: string;
+  /**
+   * The user associated with this avatar
+   */
+  user?: UserData;
+  /**
+   * Variants of how the avatar will be portrayed
+   */
+  variant?: 'icon' | 'initials' | 'image';
+}
+
+function getInitials(fromName: string): string {
+  /**
+   * Scenarios:
+   * - User's name is spelled as first and last name: John Smith
+   * - User's name has a middle name or initial: John C. Smith
+   * - User's Name has dashes in it
+   */
+  return fromName
+    .split(' ')
+    .map((part) => part[0])
+    .reduce(
+      (prev, curr, idx, arr) =>
+        idx === 0 || idx === arr.length - 1 ? prev + curr : prev,
+      '',
+    )
+    .toUpperCase();
+}
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {Avatar} from "@chanzuckerberg/eds";`
+ *
+ * Representation of a single, unique user, keyed by the user name
+ */
+export const Avatar = ({
+  ariaLabel,
+  as: TagName = 'div' as const,
+  className,
+  shape = 'circle',
+  size = 'md',
+  user,
+  variant = 'initials',
+  src,
+  ...other
+}: Props) => {
+  const componentClassName = clsx(
+    styles['avatar'],
+    shape && styles[`avatar--${shape}`],
+    size && styles[`avatar--${size}`],
+    variant && styles[`avatar--${variant}`],
+    className,
+  );
+
+  const descriptiveLabel =
+    ariaLabel ??
+    `Avatar for ${!user && 'unknown'} user ${user?.fullName || ''}`;
+
+  // when not a button, add some props to allow focus states
+  const a11yProps = TagName === 'button' ? {} : { role: 'button', tabIndex: 0 };
```

Comment:
What's the scenario for using a div vs button?

Wondering if this should either be:
- Always a div, since I don't see an `onClick` handler or other interaction?
- Always a button, since we're giving it a button role anyway?

Reply:
There are these two cases:

- an avatar can appear as an inert component in things like cards, or profiles (where there is no interaction)
- as an actual button (imagine a profile menu, where clicking the avatar shows a `Menu`'s content below)

Now that I think about it, in the former case it wouldn't need such a role or tab index b/c there's nothing to do ... its essentially an image then. This was initially to ensure focus state was working but should be removed now

Reply:
@ahuth it may be true that it should never be a button, and cases where it is clickable would be wrapped by a button? maybe I should just take this part out for now, and keep it simple

Reply:
I personally think avatar should beinert, and if interactive functionality is needed, it will be wrapped by an appropriate element.
However, this does complicate focus states, but using <Button> will assume to better fit target size requirements

Reply:
Not sure of the right answer, but maybe it's best to start with the simpler solution. Can add in a button later if we think it's better.

Reply:
@jinlee93 @ahuth Agreed. I thought about it some more , and i think what's happening here is actually more like "an avatar can be wrapped by an 'invisible' button" where the button itself has the focus ring instead of the avatar... I'm gonna take this out and let the avatar be more image-like.

There are actually 1-2 other things behaving like this in EDS already, and I've seen this variant in other design systems i've worked on, so i will try some things out in a follow-up